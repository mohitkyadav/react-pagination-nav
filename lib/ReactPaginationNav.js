module.exports=function(e){var t={};function n(r){if(t[r])return t[r].exports;var o=t[r]={i:r,l:!1,exports:{}};return e[r].call(o.exports,o,o.exports,n),o.l=!0,o.exports}return n.m=e,n.c=t,n.d=function(e,t,r){n.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:r})},n.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},n.t=function(e,t){if(1&t&&(e=n(e)),8&t)return e;if(4&t&&"object"==typeof e&&e&&e.__esModule)return e;var r=Object.create(null);if(n.r(r),Object.defineProperty(r,"default",{enumerable:!0,value:e}),2&t&&"string"!=typeof e)for(var o in e)n.d(r,o,function(t){return e[t]}.bind(null,o));return r},n.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return n.d(t,"a",t),t},n.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},n.p="",n(n.s=0)}([function(e,t,n){"use strict";Object.defineProperty(t,"__esModule",{value:!0});var r=n(1),o=u(r),a=u(n(4));function u(e){return e&&e.__esModule?e:{default:e}}n(5);var i=function(e){var t=e.className,n=e.goToPreviousPage,r=e.pageCount,u=e.currentPage,i=e.goToPage,c=e.goToNextPage,l=e.visiblePages,f=void 0===l?5:l,s=e.isPreviousBtnHidden,p=e.isNextBtnHidden,d=e.PrevNextButton,v=e.PageButton,y=parseInt(f,10)%2==0?parseInt(f,10)+1:parseInt(f,10),m=y/2;return o.default.createElement("div",{className:(0,a.default)("react-pagination-nav",t)},!s&&o.default.createElement(d,{direction:"prev",onClick:n}),o.default.createElement("div",{className:"react-pagination-nav__page-list"},Array(r).fill().map((function(e,t){return(Math.abs(u-1-t)<m||u<m&&t<y||r-u<m&&Math.abs(u-1-t)<y-(r-u))&&o.default.createElement(v,{page:t+1,active:u===t+1,onClick:function(){return i(t+1)}})}))),!p&&o.default.createElement(d,{direction:"next",onClick:c}))},c=(0,r.memo)((function(e){var t=e.direction,n=e.onClick;return o.default.createElement("button",{className:"react-pagination-nav__"+t+"-page react-pagination-nav__button",onClick:n,title:"Go to "+t+" page","aria-label":"Go to "+t+" page"},"prev"===t?"<":">")})),l=(0,r.memo)((function(e){var t=e.page,n=e.active,r=e.onClick;return o.default.createElement("button",{key:t,className:(0,a.default)("react-pagination-nav__page-number react-pagination-nav__button",n&&"react-pagination-nav__button__active"),onClick:r},t)}));i.defaultProps={PrevNextButton:c,PageButton:l},t.default=i},function(e,t,n){"use strict";e.exports=n(2)},function(e,t,n){"use strict";
/** @license React v16.13.1
 * react.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */var r=n(3),o="function"==typeof Symbol&&Symbol.for,a=o?Symbol.for("react.element"):60103,u=o?Symbol.for("react.portal"):60106,i=o?Symbol.for("react.fragment"):60107,c=o?Symbol.for("react.strict_mode"):60108,l=o?Symbol.for("react.profiler"):60114,f=o?Symbol.for("react.provider"):60109,s=o?Symbol.for("react.context"):60110,p=o?Symbol.for("react.forward_ref"):60112,d=o?Symbol.for("react.suspense"):60113,v=o?Symbol.for("react.memo"):60115,y=o?Symbol.for("react.lazy"):60116,m="function"==typeof Symbol&&Symbol.iterator;function b(e){for(var t="https://reactjs.org/docs/error-decoder.html?invariant="+e,n=1;n<arguments.length;n++)t+="&args[]="+encodeURIComponent(arguments[n]);return"Minified React error #"+e+"; visit "+t+" for the full message or use the non-minified dev environment for full errors and additional helpful warnings."}var h={isMounted:function(){return!1},enqueueForceUpdate:function(){},enqueueReplaceState:function(){},enqueueSetState:function(){}},g={};function _(e,t,n){this.props=e,this.context=t,this.refs=g,this.updater=n||h}function x(){}function j(e,t,n){this.props=e,this.context=t,this.refs=g,this.updater=n||h}_.prototype.isReactComponent={},_.prototype.setState=function(e,t){if("object"!=typeof e&&"function"!=typeof e&&null!=e)throw Error(b(85));this.updater.enqueueSetState(this,e,t,"setState")},_.prototype.forceUpdate=function(e){this.updater.enqueueForceUpdate(this,e,"forceUpdate")},x.prototype=_.prototype;var S=j.prototype=new x;S.constructor=j,r(S,_.prototype),S.isPureReactComponent=!0;var w={current:null},C=Object.prototype.hasOwnProperty,O={key:!0,ref:!0,__self:!0,__source:!0};function k(e,t,n){var r,o={},u=null,i=null;if(null!=t)for(r in void 0!==t.ref&&(i=t.ref),void 0!==t.key&&(u=""+t.key),t)C.call(t,r)&&!O.hasOwnProperty(r)&&(o[r]=t[r]);var c=arguments.length-2;if(1===c)o.children=n;else if(1<c){for(var l=Array(c),f=0;f<c;f++)l[f]=arguments[f+2];o.children=l}if(e&&e.defaultProps)for(r in c=e.defaultProps)void 0===o[r]&&(o[r]=c[r]);return{$$typeof:a,type:e,key:u,ref:i,props:o,_owner:w.current}}function P(e){return"object"==typeof e&&null!==e&&e.$$typeof===a}var E=/\/+/g,R=[];function $(e,t,n,r){if(R.length){var o=R.pop();return o.result=e,o.keyPrefix=t,o.func=n,o.context=r,o.count=0,o}return{result:e,keyPrefix:t,func:n,context:r,count:0}}function M(e){e.result=null,e.keyPrefix=null,e.func=null,e.context=null,e.count=0,10>R.length&&R.push(e)}function N(e,t,n){return null==e?0:function e(t,n,r,o){var i=typeof t;"undefined"!==i&&"boolean"!==i||(t=null);var c=!1;if(null===t)c=!0;else switch(i){case"string":case"number":c=!0;break;case"object":switch(t.$$typeof){case a:case u:c=!0}}if(c)return r(o,t,""===n?"."+A(t,0):n),1;if(c=0,n=""===n?".":n+":",Array.isArray(t))for(var l=0;l<t.length;l++){var f=n+A(i=t[l],l);c+=e(i,f,r,o)}else if(null===t||"object"!=typeof t?f=null:f="function"==typeof(f=m&&t[m]||t["@@iterator"])?f:null,"function"==typeof f)for(t=f.call(t),l=0;!(i=t.next()).done;)c+=e(i=i.value,f=n+A(i,l++),r,o);else if("object"===i)throw r=""+t,Error(b(31,"[object Object]"===r?"object with keys {"+Object.keys(t).join(", ")+"}":r,""));return c}(e,"",t,n)}function A(e,t){return"object"==typeof e&&null!==e&&null!=e.key?function(e){var t={"=":"=0",":":"=2"};return"$"+(""+e).replace(/[=:]/g,(function(e){return t[e]}))}(e.key):t.toString(36)}function T(e,t){e.func.call(e.context,t,e.count++)}function I(e,t,n){var r=e.result,o=e.keyPrefix;e=e.func.call(e.context,t,e.count++),Array.isArray(e)?B(e,r,n,(function(e){return e})):null!=e&&(P(e)&&(e=function(e,t){return{$$typeof:a,type:e.type,key:t,ref:e.ref,props:e.props,_owner:e._owner}}(e,o+(!e.key||t&&t.key===e.key?"":(""+e.key).replace(E,"$&/")+"/")+n)),r.push(e))}function B(e,t,n,r,o){var a="";null!=n&&(a=(""+n).replace(E,"$&/")+"/"),N(e,I,t=$(t,a,r,o)),M(t)}var U={current:null};function L(){var e=U.current;if(null===e)throw Error(b(321));return e}var q={ReactCurrentDispatcher:U,ReactCurrentBatchConfig:{suspense:null},ReactCurrentOwner:w,IsSomeRendererActing:{current:!1},assign:r};t.Children={map:function(e,t,n){if(null==e)return e;var r=[];return B(e,r,null,t,n),r},forEach:function(e,t,n){if(null==e)return e;N(e,T,t=$(null,null,t,n)),M(t)},count:function(e){return N(e,(function(){return null}),null)},toArray:function(e){var t=[];return B(e,t,null,(function(e){return e})),t},only:function(e){if(!P(e))throw Error(b(143));return e}},t.Component=_,t.Fragment=i,t.Profiler=l,t.PureComponent=j,t.StrictMode=c,t.Suspense=d,t.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED=q,t.cloneElement=function(e,t,n){if(null==e)throw Error(b(267,e));var o=r({},e.props),u=e.key,i=e.ref,c=e._owner;if(null!=t){if(void 0!==t.ref&&(i=t.ref,c=w.current),void 0!==t.key&&(u=""+t.key),e.type&&e.type.defaultProps)var l=e.type.defaultProps;for(f in t)C.call(t,f)&&!O.hasOwnProperty(f)&&(o[f]=void 0===t[f]&&void 0!==l?l[f]:t[f])}var f=arguments.length-2;if(1===f)o.children=n;else if(1<f){l=Array(f);for(var s=0;s<f;s++)l[s]=arguments[s+2];o.children=l}return{$$typeof:a,type:e.type,key:u,ref:i,props:o,_owner:c}},t.createContext=function(e,t){return void 0===t&&(t=null),(e={$$typeof:s,_calculateChangedBits:t,_currentValue:e,_currentValue2:e,_threadCount:0,Provider:null,Consumer:null}).Provider={$$typeof:f,_context:e},e.Consumer=e},t.createElement=k,t.createFactory=function(e){var t=k.bind(null,e);return t.type=e,t},t.createRef=function(){return{current:null}},t.forwardRef=function(e){return{$$typeof:p,render:e}},t.isValidElement=P,t.lazy=function(e){return{$$typeof:y,_ctor:e,_status:-1,_result:null}},t.memo=function(e,t){return{$$typeof:v,type:e,compare:void 0===t?null:t}},t.useCallback=function(e,t){return L().useCallback(e,t)},t.useContext=function(e,t){return L().useContext(e,t)},t.useDebugValue=function(){},t.useEffect=function(e,t){return L().useEffect(e,t)},t.useImperativeHandle=function(e,t,n){return L().useImperativeHandle(e,t,n)},t.useLayoutEffect=function(e,t){return L().useLayoutEffect(e,t)},t.useMemo=function(e,t){return L().useMemo(e,t)},t.useReducer=function(e,t,n){return L().useReducer(e,t,n)},t.useRef=function(e){return L().useRef(e)},t.useState=function(e){return L().useState(e)},t.version="16.13.1"},function(e,t,n){"use strict";
/*
object-assign
(c) Sindre Sorhus
@license MIT
*/var r=Object.getOwnPropertySymbols,o=Object.prototype.hasOwnProperty,a=Object.prototype.propertyIsEnumerable;function u(e){if(null==e)throw new TypeError("Object.assign cannot be called with null or undefined");return Object(e)}e.exports=function(){try{if(!Object.assign)return!1;var e=new String("abc");if(e[5]="de","5"===Object.getOwnPropertyNames(e)[0])return!1;for(var t={},n=0;n<10;n++)t["_"+String.fromCharCode(n)]=n;if("0123456789"!==Object.getOwnPropertyNames(t).map((function(e){return t[e]})).join(""))return!1;var r={};return"abcdefghijklmnopqrst".split("").forEach((function(e){r[e]=e})),"abcdefghijklmnopqrst"===Object.keys(Object.assign({},r)).join("")}catch(e){return!1}}()?Object.assign:function(e,t){for(var n,i,c=u(e),l=1;l<arguments.length;l++){for(var f in n=Object(arguments[l]))o.call(n,f)&&(c[f]=n[f]);if(r){i=r(n);for(var s=0;s<i.length;s++)a.call(n,i[s])&&(c[i[s]]=n[i[s]])}}return c}},function(e,t,n){var r;
/*!
  Copyright (c) 2017 Jed Watson.
  Licensed under the MIT License (MIT), see
  http://jedwatson.github.io/classnames
*/!function(){"use strict";var n={}.hasOwnProperty;function o(){for(var e=[],t=0;t<arguments.length;t++){var r=arguments[t];if(r){var a=typeof r;if("string"===a||"number"===a)e.push(r);else if(Array.isArray(r)&&r.length){var u=o.apply(null,r);u&&e.push(u)}else if("object"===a)for(var i in r)n.call(r,i)&&r[i]&&e.push(i)}}return e.join(" ")}e.exports?(o.default=o,e.exports=o):void 0===(r=function(){return o}.apply(t,[]))||(e.exports=r)}()},function(e,t,n){var r=n(6),o=n(7);"string"==typeof(o=o.__esModule?o.default:o)&&(o=[[e.i,o,""]]);var a={insert:"head",singleton:!1};r(o,a);e.exports=o.locals||{}},function(e,t,n){"use strict";var r,o=function(){return void 0===r&&(r=Boolean(window&&document&&document.all&&!window.atob)),r},a=function(){var e={};return function(t){if(void 0===e[t]){var n=document.querySelector(t);if(window.HTMLIFrameElement&&n instanceof window.HTMLIFrameElement)try{n=n.contentDocument.head}catch(e){n=null}e[t]=n}return e[t]}}(),u=[];function i(e){for(var t=-1,n=0;n<u.length;n++)if(u[n].identifier===e){t=n;break}return t}function c(e,t){for(var n={},r=[],o=0;o<e.length;o++){var a=e[o],c=t.base?a[0]+t.base:a[0],l=n[c]||0,f="".concat(c," ").concat(l);n[c]=l+1;var s=i(f),p={css:a[1],media:a[2],sourceMap:a[3]};-1!==s?(u[s].references++,u[s].updater(p)):u.push({identifier:f,updater:m(p,t),references:1}),r.push(f)}return r}function l(e){var t=document.createElement("style"),r=e.attributes||{};if(void 0===r.nonce){var o=n.nc;o&&(r.nonce=o)}if(Object.keys(r).forEach((function(e){t.setAttribute(e,r[e])})),"function"==typeof e.insert)e.insert(t);else{var u=a(e.insert||"head");if(!u)throw new Error("Couldn't find a style target. This probably means that the value for the 'insert' parameter is invalid.");u.appendChild(t)}return t}var f,s=(f=[],function(e,t){return f[e]=t,f.filter(Boolean).join("\n")});function p(e,t,n,r){var o=n?"":r.media?"@media ".concat(r.media," {").concat(r.css,"}"):r.css;if(e.styleSheet)e.styleSheet.cssText=s(t,o);else{var a=document.createTextNode(o),u=e.childNodes;u[t]&&e.removeChild(u[t]),u.length?e.insertBefore(a,u[t]):e.appendChild(a)}}function d(e,t,n){var r=n.css,o=n.media,a=n.sourceMap;if(o?e.setAttribute("media",o):e.removeAttribute("media"),a&&btoa&&(r+="\n/*# sourceMappingURL=data:application/json;base64,".concat(btoa(unescape(encodeURIComponent(JSON.stringify(a))))," */")),e.styleSheet)e.styleSheet.cssText=r;else{for(;e.firstChild;)e.removeChild(e.firstChild);e.appendChild(document.createTextNode(r))}}var v=null,y=0;function m(e,t){var n,r,o;if(t.singleton){var a=y++;n=v||(v=l(t)),r=p.bind(null,n,a,!1),o=p.bind(null,n,a,!0)}else n=l(t),r=d.bind(null,n,t),o=function(){!function(e){if(null===e.parentNode)return!1;e.parentNode.removeChild(e)}(n)};return r(e),function(t){if(t){if(t.css===e.css&&t.media===e.media&&t.sourceMap===e.sourceMap)return;r(e=t)}else o()}}e.exports=function(e,t){(t=t||{}).singleton||"boolean"==typeof t.singleton||(t.singleton=o());var n=c(e=e||[],t);return function(e){if(e=e||[],"[object Array]"===Object.prototype.toString.call(e)){for(var r=0;r<n.length;r++){var o=i(n[r]);u[o].references--}for(var a=c(e,t),l=0;l<n.length;l++){var f=i(n[l]);0===u[f].references&&(u[f].updater(),u.splice(f,1))}n=a}}}},function(e,t,n){(t=n(8)(!1)).push([e.i,".react-pagination-nav{display:flex}.react-pagination-nav__page-list{display:flex}.react-pagination-nav__button{margin-right:10px;border:none;background-color:transparent;border:2px solid transparent;border-radius:2px;color:#fff;font-size:16px;width:25px;height:25px;transition:all .3s ease-in-out}.react-pagination-nav__button__active{background-color:#6495ed}.react-pagination-nav__button:hover,.react-pagination-nav__button:focus{cursor:pointer;border:2px solid #6495ed;outline:none}",""]),e.exports=t},function(e,t,n){"use strict";e.exports=function(e){var t=[];return t.toString=function(){return this.map((function(t){var n=function(e,t){var n=e[1]||"",r=e[3];if(!r)return n;if(t&&"function"==typeof btoa){var o=(u=r,i=btoa(unescape(encodeURIComponent(JSON.stringify(u)))),c="sourceMappingURL=data:application/json;charset=utf-8;base64,".concat(i),"/*# ".concat(c," */")),a=r.sources.map((function(e){return"/*# sourceURL=".concat(r.sourceRoot||"").concat(e," */")}));return[n].concat(a).concat([o]).join("\n")}var u,i,c;return[n].join("\n")}(t,e);return t[2]?"@media ".concat(t[2]," {").concat(n,"}"):n})).join("")},t.i=function(e,n,r){"string"==typeof e&&(e=[[null,e,""]]);var o={};if(r)for(var a=0;a<this.length;a++){var u=this[a][0];null!=u&&(o[u]=!0)}for(var i=0;i<e.length;i++){var c=[].concat(e[i]);r&&o[c[0]]||(n&&(c[2]?c[2]="".concat(n," and ").concat(c[2]):c[2]=n),t.push(c))}},t}}]);