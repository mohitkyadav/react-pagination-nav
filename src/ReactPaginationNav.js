import React, { memo } from 'react'
import cn from 'classnames'
import PropTypes from 'prop-types'

import './index.scss'

/** props
 *
 * goToPreviousPage : called when pressing the left arrow button
 * goToNextPage     : called when pressing the right arrow button
 * goToPage         : called when pressing any of the page numbers
 * goToFirstPage    : called when pressing the "Go to first page"
 * goToLastPage     : called when pressing the "Go to last page"
 *
 * visiblePages : odd number of pages you want to be visible, default 5
 * pageCount    : total number of pages
 * currentPage  : current page number
 * isFirstBtnHidden  : If false, display "Go to first page"
 * isLastBtnHidden  : If false, display "Go to last page"
 * PrevNextButton : a button component for the prev and next buttons
 * PageButton     : a button component for the page buttons
 * theme : set colors palette
 */
const ReactPaginationNav = ({
  theme = 'dark',
  className, goToPreviousPage, pageCount, currentPage,
  goToPage, goToNextPage, visiblePages = 5,
  isPreviousBtnHidden, isNextBtnHidden,
  goToFirstPage, goToLastPage,
  isFirstBtnHidden, isLastBtnHidden,
  PageButton, PrevNextButton
}) => {
  // in case visiblePages is an even number
  const oddVisiblePages = (parseInt(visiblePages, 10) % 2) === 0
    ? parseInt(visiblePages, 10) + 1
    : parseInt(visiblePages, 10)
  const halfVisiblePages = oddVisiblePages / 2

  return (
    <div className={cn('react-pagination-nav', `react-pagination-nav--${theme}`, className)}>
      {!isFirstBtnHidden && (
        <button
          className="react-pagination-nav__first-page react-pagination-nav__button"
          onClick={() => goToFirstPage && goToFirstPage()}
          title="Go to first page"
          aria-label="Go to first page"
        >
          {'First'}
        </button>
      )}

      {!isPreviousBtnHidden && <PrevNextButton direction='prev' onClick={goToPreviousPage} />}

      <div className="react-pagination-nav__page-list">
        {
          Array(pageCount).fill().map((_, i) => {
            return (
              (
                Math.abs(currentPage - 1 - i) < halfVisiblePages ||
                (currentPage < halfVisiblePages && (i < oddVisiblePages)) ||
                (
                  pageCount - currentPage < halfVisiblePages &&
                  (Math.abs(currentPage - 1 - i) < (oddVisiblePages - (pageCount - currentPage)))
                )
              ) &&
              (
                <React.Fragment>
                  <PageButton page={i + 1} active={currentPage === i + 1} onClick={() => goToPage(i + 1)} />
                </React.Fragment>
              )
            )
          })
        }
      </div>

      {!isNextBtnHidden && <PrevNextButton direction='next' onClick={goToNextPage} />}

      {!isLastBtnHidden && (
        <button
          className="react-pagination-nav__last-page react-pagination-nav__button"
          onClick={() => goToLastPage && goToLastPage()}
          title="Go to last page"
          aria-label="Go to last page"
        >
          {'Last'}
        </button>
      )}
    </div>
  )
}

const DefaultPrevNextButton = memo(({ direction, onClick }) => {
  return <button
    className={`react-pagination-nav__${direction}-page react-pagination-nav__button`}
    onClick={onClick}
    title={`Go to ${direction} page`}
    aria-label={`Go to ${direction} page`}
  >
    {direction === 'prev' ? '<' : '>'}
  </button>
})

const DefaultPageButton = memo(({ page, active, onClick }) => {
  return <button
    key={page}
    className={cn(
      "react-pagination-nav__page-number react-pagination-nav__button",
      ((active) && 'react-pagination-nav__button__active')
    )}
    onClick={onClick}
  >
    {page}
  </button>
})

ReactPaginationNav.defaultProps = {
  PrevNextButton: DefaultPrevNextButton,
  PageButton: DefaultPageButton
}

ReactPaginationNav.propTypes = {
  PageButton: PropTypes.element,
  PrevNextButton: PropTypes.element,
  goToPreviousPage: PropTypes.func.isRequired,
  goToNextPage: PropTypes.func.isRequired,
  goToPage: PropTypes.func.isRequired,
  currentPage: PropTypes.number.isRequired,
  pageCount: PropTypes.number.isRequired,
  className: PropTypes.string,
  visiblePages: PropTypes.number,
  isPreviousBtnHidden: PropTypes.bool,
  isNextBtnHidden: PropTypes.bool,
  theme: PropTypes.string
}

export default ReactPaginationNav
