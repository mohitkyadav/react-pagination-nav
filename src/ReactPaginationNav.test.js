import React from 'react'
import { create, act } from 'react-test-renderer'
import ReactPaginationNav from './ReactPaginationNav'

const App = ({ page = 1 }) => {
  const [currentPage, setCurrentPage] = React.useState(page)
  return (
    <ReactPaginationNav
      pageCount={6}
      visiblePages={3}
      currentPage={currentPage}
      goToNextPage={() => setCurrentPage(currentPage + 1)}
      goToPreviousPage={() => setCurrentPage(currentPage - 1)}
      goToPage={(newPage) => setCurrentPage(newPage)}
    />
  )
}

const snapMatch = (component) => {
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
}

test('Nav renders and switches between pages', () => {
  let component
  act(() => {
    component = create(<App />)
  })
  const nextBtn = component.root.findByProps({ className: 'react-pagination-nav__next-page react-pagination-nav__button' })
  const prevBtn = component.root.findByProps({ className: 'react-pagination-nav__prev-page react-pagination-nav__button' })
  const otherBtn = component.root.findByProps({ children: 3 })
  snapMatch(component)

  act(() => {
    nextBtn.props.onClick()
  })
  snapMatch(component)

  act(() => {
    prevBtn.props.onClick()
  })
  snapMatch(component)

  act(() => {
    otherBtn.props.onClick()
  })
  snapMatch(component)
})
