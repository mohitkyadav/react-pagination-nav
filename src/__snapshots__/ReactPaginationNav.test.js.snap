// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders and switches between pages 1`] = `
<div
  className="react-pagination-nav"
>
  <button
    aria-label="Go to previous page"
    className="react-pagination-nav__prev-page react-pagination-nav__button"
    onClick={[Function]}
    title="Go to previous page"
  >
    &lt;
  </button>
  <div
    className="react-pagination-nav__page-list"
  >
    <button
      className="react-pagination-nav__page-number react-pagination-nav__button react-pagination-nav__button__active"
      onClick={[Function]}
    >
      1
    </button>
    <button
      className="react-pagination-nav__page-number react-pagination-nav__button "
      onClick={[Function]}
    >
      2
    </button>
    <button
      className="react-pagination-nav__page-number react-pagination-nav__button "
      onClick={[Function]}
    >
      3
    </button>
  </div>
  <button
    aria-label="Go to next page"
    className="react-pagination-nav__next-page react-pagination-nav__button"
    onClick={[Function]}
    title="Go to next page"
  >
    &gt;
  </button>
</div>
`;

exports[`Nav renders and switches between pages 2`] = `
<div
  className="react-pagination-nav"
>
  <button
    aria-label="Go to previous page"
    className="react-pagination-nav__prev-page react-pagination-nav__button"
    onClick={[Function]}
    title="Go to previous page"
  >
    &lt;
  </button>
  <div
    className="react-pagination-nav__page-list"
  >
    <button
      className="react-pagination-nav__page-number react-pagination-nav__button "
      onClick={[Function]}
    >
      1
    </button>
    <button
      className="react-pagination-nav__page-number react-pagination-nav__button react-pagination-nav__button__active"
      onClick={[Function]}
    >
      2
    </button>
    <button
      className="react-pagination-nav__page-number react-pagination-nav__button "
      onClick={[Function]}
    >
      3
    </button>
  </div>
  <button
    aria-label="Go to next page"
    className="react-pagination-nav__next-page react-pagination-nav__button"
    onClick={[Function]}
    title="Go to next page"
  >
    &gt;
  </button>
</div>
`;

exports[`Nav renders and switches between pages 3`] = `
<div
  className="react-pagination-nav"
>
  <button
    aria-label="Go to previous page"
    className="react-pagination-nav__prev-page react-pagination-nav__button"
    onClick={[Function]}
    title="Go to previous page"
  >
    &lt;
  </button>
  <div
    className="react-pagination-nav__page-list"
  >
    <button
      className="react-pagination-nav__page-number react-pagination-nav__button react-pagination-nav__button__active"
      onClick={[Function]}
    >
      1
    </button>
    <button
      className="react-pagination-nav__page-number react-pagination-nav__button "
      onClick={[Function]}
    >
      2
    </button>
    <button
      className="react-pagination-nav__page-number react-pagination-nav__button "
      onClick={[Function]}
    >
      3
    </button>
  </div>
  <button
    aria-label="Go to next page"
    className="react-pagination-nav__next-page react-pagination-nav__button"
    onClick={[Function]}
    title="Go to next page"
  >
    &gt;
  </button>
</div>
`;

exports[`Nav renders and switches between pages 4`] = `
<div
  className="react-pagination-nav"
>
  <button
    aria-label="Go to previous page"
    className="react-pagination-nav__prev-page react-pagination-nav__button"
    onClick={[Function]}
    title="Go to previous page"
  >
    &lt;
  </button>
  <div
    className="react-pagination-nav__page-list"
  >
    <button
      className="react-pagination-nav__page-number react-pagination-nav__button "
      onClick={[Function]}
    >
      2
    </button>
    <button
      className="react-pagination-nav__page-number react-pagination-nav__button react-pagination-nav__button__active"
      onClick={[Function]}
    >
      3
    </button>
    <button
      className="react-pagination-nav__page-number react-pagination-nav__button "
      onClick={[Function]}
    >
      4
    </button>
  </div>
  <button
    aria-label="Go to next page"
    className="react-pagination-nav__next-page react-pagination-nav__button"
    onClick={[Function]}
    title="Go to next page"
  >
    &gt;
  </button>
</div>
`;
